pkg/store/prometheus.go

```diff
@@ -709,6 +709,11 @@ func (p *PrometheusStore) LabelValues(ctx context.Context, r *storepb.LabelValue
 
 	// First check for matching external label which has priority.
 	if l := extLset.Get(r.Label); l != "" {
+		for _, m := range matchers {
+			if !m.Matches(l) {
+				return &storepb.LabelValuesResponse{Values: nil}, nil
```

Comment:
I don't think we need to explicitly set `Values: nil` because that's the zero value for that field, no?

Reply:
Yeah thats true!

Reply:
Done ( sorry for the delay ); i also changed the other instances of this to be uniform.